### PR TITLE
feat: implement profile-based configuration with connection string auth

### PR DESCRIPTION
**Pull request by dawidpereira**
- Add multi-environment profile system (dev/staging/prod) with isolated credentials
- Implement connection string authentication option in setup wizard
- Add embedded default configurations (config.default.toml, keys.default.toml)
- Create ProfileCache with 30s TTL for performance optimization
- Add comprehensive security validation preventing path traversal attacks
- Consolidate documentation removing 40% duplication across guides
- Support all authentication methods: device code, client credentials, connection string
- Add interactive setup wizard with input validation and Azure Portal guidance
- Implement secure .env file storage with restrictive permissions (600)
- Create 300+ tests covering security edge cases and functionality

### CODE REVIEW

File: ui/src/config/mod.rs [hunk 1]
```diff
@@ -76,38 +82,265 @@ static CONFIG: std::sync::OnceLock<ConfigLoadResult> = std::sync::OnceLock::new(
...
+    if let Ok(config_dir) = setup::get_config_dir() {
+        let profile_env_path = config_dir.join("profiles").join("default").join(".env");
+        if profile_env_path.exists() {
+            dotenv::from_path(profile_env_path).ok();
+        }
+    }
+
+    // Also try loading .env from current directory (for backward compatibility)
+    dotenv::from_path(".env").ok();
+
```
[the remark this answers is not in the document]
Comment (dawidpereira) [reply ui/src/config/mod.rs:188]: Agree, will be removed.
